### DISCUSSION TITLE
technical zsh question

### DISCUSSION THREAD

**Discussion by ghost** [ohmyzsh/ohmyzsh#11055] [Help] [answered]
today i was looking for a feature that omz enables by default.
basically it makes searching through history using arrows much easier and removes all the duplicates. For example if i had the following in the history:
```
vim script.sh
./script.sh
./script.sh
./script.sh
```
it would remove two `./script.sh` strings when searching and only 2 times should be pressed to reach `vim script.sh` command.
After some research and a bit of reverse engineering i found that command `echoti smkx` in [`$ZSH/lib/key-bindings.zsh`](https://github.com/ohmyzsh/ohmyzsh/blob/master/lib/key-bindings.zsh#L9) makes this _feature_ possible(_at least on my system_).
I would like to know what this command does and what it was intended for and if it works like on my system.

**Comment by rwmitchell** [comment, marked as answer]
The zsh command to ignore history duplicates is:

```setopt histignoredups```

**Reply by ghost** [reply]
Thanks! It looks like i found some weird way lol
